tests/network/vmi_istio.go

```diff
@@ -0,0 +1,297 @@
+	)
+	// Istio-proxy requires service account token to be mounted
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	BeforeEach(func() {
+		if !istioServiceMeshDeployed() {
+			Skip("Istio service mesh is required for service-mesh tests to run")
+		}
+	})
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+	network := &v1.Network{
+		Name: "default",
+		NetworkSource: v1.NetworkSource{
+			Pod: &v1.PodNetwork{},
+		},
+	}
+
+	Context("Virtual Machine with masquerade interface and explicitly specified ports", func() {
+		verifyClientServerConnectivity := func(serverVMI *v1.VirtualMachineInstance, targetPort int) (*batchv1.Job, error) {
+			By("Starting HTTP Server")
+			tests.StartHTTPServer(vmi, targetPort)
+
+			By("Getting back the VMI IP")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmiIP := vmi.Status.Interfaces[0].IP
+
+			By("Running job to send a request to the server")
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+			tests.AddLabelToNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel, "enabled")
+			ports := []v1.Port{
+				{Port: svcDeclaredTestPort},
+				{Port: svcUndeclaredTestPort},
+			}
+			vmi = newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(
+				vmiAppSelector,
+				network,
+				ports,
+			)
+			service = newServiceWithAppSelector(
+				fmt.Sprintf("%s-service", vmi.Name),
+				vmi.Namespace,
+				vmiAppSelector,
+			)
+
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+
+		})
+		AfterEach(func() {
+			tests.RemoveLabelFromNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel)
+			err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.Name, &metav1.DeleteOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Delete(context.TODO(), service.Name, metav1.DeleteOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		Describe("Inbound traffic", func() {
+			table.DescribeTable("request to VMI", func(port int) {
+				var job *batchv1.Job
+				job, err = verifyClientServerConnectivity(vmi, port)
```

Comment:
As mentioned in this comment: https://github.com/kubevirt/kubevirt/pull/5200#discussion_r614976212
Added AfterEach to remove the Job with its children

Reply:
Sorry I missed you tagged me on the comment. We decided to use the `BeforeTestCleanup`, since currently that's what most of out tests are using.
I don't remember the exact reason why removing just what you created in the after each caused us issued. @EdDev  do you remember?

Reply:
It is unfortunate, but the current testing infrastructure we have reports the current state of the system (logs, entities states, etc) at a stage that occurs **after** the teardown is executed. As an outcome, on failure, cleanups that are done in the teardown will not be collected by the reporter and we will lack some information for debugging.